extra/3-magic-8-ball.js

```diff
@@ -13,40 +13,55 @@
   and have different levels of positivity or negativity.
 
   Below are the possible answers:
-
-  ## Very positive
-    It is certain.
-    It is decidedly so.
-    Without a doubt.
-    Yes - definitely.
-    You may rely on it.
-
-  ## Positive
-    As I see it, yes.
-    Most likely.
-    Outlook good.
-    Yes.
-    Signs point to yes.
-
-  ## Negative
-    Reply hazy, try again.
-    Ask again later.
-    Better not tell you now.
-    Cannot predict now.
-    Concentrate and ask again.
-
-  ## Very negative
-    Don't count on it.
-    My reply is no.
-    My sources say no.
-    Outlook not so good.
-    Very doubtful.
 */
 
+  let veryPositiveAnswers = [
+   "It is certain.",
+    "It is decidedly so.",
+    "Without a doubt.",
+    "Yes - definitely.",
+    "You may rely on it."
+  ];
+
+  let positiveAnswers= [
+    "As I see it, yes.",
+    "Most likely.",
+    "Outlook good.",
+    "Yes.",
+    "Signs point to yes.",
+  ];
+
+  let negativeAnswers = [
+    "Reply hazy, try again.",
+    "Ask again later.",
+    "Better not tell you now.",
+    "Cannot predict now.",
+    "Concentrate and ask again.",
+  ];
+
+  let veryNegativeAnswers = [
+    "Don't count on it.",
+    "My reply is no.",
+    "My sources say no.",
+    "Outlook not so good.",
+    "Very doubtful.",
+  ];
+
+  let allAnswers = [
+    ...veryPositiveAnswers,
+   ... positiveAnswers,
+    ...negativeAnswers,
+    ...veryNegativeAnswers,
+  ];
+
+
+
 // This should log "The ball has shaken!"
 // and return the answer.
 function shakeBall() {
-  //Write your code in here
+  console.log ("The ball has shaken!");
+  let randomIndex = Math.floor(Math.random() * allAnswers.length);
+  return allAnswers [randomIndex];
```

Comment:
well done! Just be careful here as there shouldn't be a space between `allAnswers` and `[randomIndex]`, it should be `allAnswers[randomIndex]`